File: app/src/main/java/top/easelink/lcg/ui/about/view/AboutFragment.java
package top.easelink.lcg.ui.about.view;

import android.os.Bundle;
import androidx.annotation.Nullable;
import androidx.lifecycle.ViewModelProviders;
import top.easelink.framework.BR;
import top.easelink.framework.base.BaseFragment;
import top.easelink.lcg.R;
import top.easelink.lcg.databinding.FragmentAboutBinding;
import top.easelink.lcg.ui.about.viewmodel.AboutViewModel;

public class AboutFragment extends BaseFragment<FragmentAboutBinding, AboutViewModel> implements AboutNavigator {

    public static final String TAG = AboutFragment.class.getSimpleName();

    private AboutViewModel mAboutViewModel;

    public static AboutFragment newInstance() {
        Bundle args = new Bundle();
        AboutFragment fragment = new AboutFragment();
        fragment.setArguments(args);
        return fragment;
    }

    @Override
    public int getBindingVariable() {
        return BR.viewModel;
    }

    @Override
    public int getLayoutId() {
        return R.layout.fragment_about;
    }

    @Override
    public AboutViewModel getViewModel() {
        mAboutViewModel = ViewModelProviders.of(this).get(AboutViewModel.class);
        return mAboutViewModel;
    }

    @Override
    public void goBack() {
        getBaseActivity().onFragmentDetached(TAG);
    }

    @Override
    public void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        mAboutViewModel.setNavigator(this);
    }
}

File: app/src/main/java/top/easelink/lcg/ui/main/about/view/AboutFragment.kt
package top.easelink.lcg.ui.main.about.view

import android.content.Intent
import android.os.Bundle
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import kotlinx.android.synthetic.main.fragment_about.*
import top.easelink.framework.topbase.ControllableFragment
import top.easelink.framework.topbase.TopFragment
import top.easelink.lcg.R
import top.easelink.lcg.ui.webview.view.WebViewActivity
import java.util.*


class AboutFragment : TopFragment(), ControllableFragment {

    override fun onCreateView(
        inflater: LayoutInflater,
        container: ViewGroup?,
        savedInstanceState: Bundle?
    ): View? {
        return inflater.inflate(R.layout.fragment_about, container, false)
    }

    override fun isControllable(): Boolean {
        return true
    }

    override fun onViewCreated(
        view: View,
        savedInstanceState: Bundle?
    ) {
        super.onViewCreated(view, savedInstanceState)
        syncAuthorState()
        github_url.setOnClickListener {
            WebViewActivity.startWebViewWith(getString(R.string.github_url), it.context)
        }
        author_email.setOnClickListener {
            val intent = Intent(Intent.ACTION_SEND)
            intent.type = "text/plain"
            intent.putExtra(Intent.EXTRA_EMAIL, getString(R.string.author_email))
            intent.putExtra(Intent.EXTRA_SUBJECT, "问题反馈")
//            intent.putExtra(Intent.EXTRA_TEXT, "")
            startActivity(Intent.createChooser(intent, "发送反馈邮件"))
        }
    }

    private fun syncAuthorState() {
        val hour = Calendar.getInstance()[Calendar.HOUR_OF_DAY]
        when {
            hour < 7 -> {
                me.setAnimation(R.raw.moon_stars)
            }
            hour < 12 -> {
                me.setAnimation(R.raw.personal_mac_daytime)
            }
            hour == 12 -> {
                me.setAnimation(R.raw.sun)
            }
            hour < 18 -> {
                me.setAnimation(R.raw.personal_phone_daytime)
            }
            hour < 22 -> {
                me.setAnimation(R.raw.personal_mac_night)
            }
            else -> {
                me.setAnimation(R.raw.personal_phone_night)
            }
        }
    }

    companion object {
        @JvmStatic
        fun newInstance(): AboutFragment {
            val args = Bundle()
            val fragment = AboutFragment()
            fragment.arguments = args
            return fragment
        }
    }
}